### PR TITLE
feat(condensed horizontal nav): updates condensed nav to match Core horizontal & vertical nav

### PR DESCRIPTION
**Pull request by priley86**
**What**:
Changes condensed nav implementation to match Core's new `pf-m-tall` class. The nav is no longer "tall" by default and this class is only applied when using the `useCondensed` feature on desktop. 
See: https://github.com/patternfly/patternfly-next/pull/884

This also introduces the need to consume the `isTall` prop within the `NavList` for Horizontal Navs, thus I have introduced a `PageContext` which will provide this context to any child elements needing this prop. If the `Nav` is wrapped with a `Page`, it will consume the prop from PageContext instead. This same context can also now be used in the `PageHeader` as well. 

**Additional issues**:

### CODE REVIEW

File: packages/patternfly-4/react-core/src/layouts/Page/Page.js [hunk 1]
```diff
@@ -33,54 +34,71 @@ const defaultProps = {
...
     const { useCondensed } = this.props;
     if (useCondensed) {
       this.mainRef.current.removeEventListener('scroll', this.handleScroll);
+      window.removeEventListener('resize', this.handleResize);
     }
   }
 
+  // only enable tall header if the user is using the `useCondensed` header feature and we are on desktop
+  isTall = props =>
+    typeof window !== 'undefined' && window.innerWidth >= parseInt(breakpointMd.value, 10) && props.useCondensed;
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/layouts/Page/Page.js]: Nit: Maybe check props.useCondensed first

File: packages/patternfly-4/react-core/src/layouts/Page/Page.js [hunk 2]
```diff
@@ -33,54 +34,71 @@ const defaultProps = {
...
-      const mainPosition = main.scrollTop;
-      if (mainPosition > scrollingDistance && !isCondensed) {
-        this.setState({ isCondensed: true });
-      } else if (mainPosition < scrollingDistance && isCondensed) {
-        this.setState({ isCondensed: false });
-      }
-    });
+    const { scrollingDistance, useCondensed } = this.props;
+    const { isTall } = this.state;
+    if (useCondensed && window.innerWidth >= parseInt(breakpointMd.value, 10)) {
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/layouts/Page/Page.js]:
```suggestion
    if (this.isTall(this.props)) {
```

File: packages/patternfly-4/react-core/src/layouts/Page/Page.js (hunk 1 above)
Comment (priley86) [reply packages/patternfly-4/react-core/src/layouts/Page/Page.js]: giving this a spin..

File: packages/patternfly-4/react-core/src/layouts/Page/Page.js (hunk 2 above)
Comment (priley86) [reply packages/patternfly-4/react-core/src/layouts/Page/Page.js]: 👍

File: packages/patternfly-4/react-core/src/components/Nav/NavList.js [hunk 3]
```diff
@@ -3,24 +3,27 @@ import PropTypes from 'prop-types';
...
 
 const propTypes = {
   /** Indicates the list type. */
   variant: PropTypes.oneOf(Object.values(NavVariants)),
   /** Children nodes */
   children: PropTypes.node,
   /** Additional classes added to the list */
-  className: PropTypes.string
+  className: PropTypes.string,
+  /** nav should have tall modifer */
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/components/Nav/NavList.js]: Can you make this comment more descriptive? Maybe something like: Horizontal nav only: If true makes the nav taller

File: packages/patternfly-4/react-core/src/layouts/Page/Page.js [hunk 4]
```diff
@@ -18,7 +19,7 @@ const propTypes = {
   header: PropTypes.node,
   /** Sidebar component for a side nav (e.g. <PageSidebar />) */
   sidebar: PropTypes.node,
-  /** condensed header on scroll */
+  /** enable condensed header on scroll feature (for desktop only) */
```
[the remark this answers is not in the document]
Comment (priley86) [reply packages/patternfly-4/react-core/src/layouts/Page/Page.js]: added comments

File: packages/patternfly-4/react-core/src/components/Nav/NavList.js (hunk 3 above)
Comment (priley86) [reply packages/patternfly-4/react-core/src/components/Nav/NavList.js]: added comments

**Pull request merged**